server/src/main/java/org/opensearch/OpenSearchException.java

```diff
@@ -594,16 +594,14 @@ public static void generateFailureXContent(XContentBuilder builder, Params param
 
         // Render the exception with a simple message
         if (detailed == false) {
-            String message = "No OpenSearchException found";
             Throwable t = e;
             for (int counter = 0; counter < 10 && t != null; counter++) {
                 if (t instanceof OpenSearchException) {
-                    message = t.getClass().getSimpleName() + "[" + t.getMessage() + "]";
                     break;
                 }
                 t = t.getCause();
             }
-            builder.field(ERROR, message);
+            builder.field(ERROR, ExceptionsHelper.summaryMessage(t));
```

Comment:
we should be passing in `e` since `t` is null... this is causing other confusing `Internal failure` exceptions even when the original exception was an `InternalArgumentException`.  see this [build failure](https://build.ci.opensearch.org/job/gradle-check/4166/)

Fixing in #4702